### PR TITLE
Adds config option exclude_errors (#342)

### PR DESCRIPTION
**Pull request by joshuap**
This branch restores #342. See the original PR for summary and initial discussion.

Let's continue discussion on this new PR.

### CODE REVIEW

File: lib/honeybadger.ex
Comment (joshuap) [inline lib/honeybadger.ex:296]: Per @sorentwo ([link](https://github.com/honeybadger-io/honeybadger-elixir/pull/442#discussion_r936572264)):

> This seems like a compile-time option rather than a runtime option (not to mention this would be a hot path for an application experiencing a rush of errors). What about pulling this up into a compile_env?

File: test/honeybadger_test.exs
Comment (joshuap) [inline test/honeybadger_test.exs:99]: Is it possible to also support error classes themselves in addition to the string `error.name`? I think some people will configure `exclude_errors: [FunctionClauseError]`.

**Review by joshuap — COMMENTED** [review]
Looking good @TraceyOnim! A few comments.

**Comment by joshuap** [comment]
This also need a [changelog entry](https://github.com/honeybadger-io/honeybadger-elixir/blob/master/CHANGELOG.md).

File: test/honeybadger_test.exs
Comment (TraceyOnim) [reply test/honeybadger_test.exs:99]: Yes, this would be possible . We might need to convert the exception classes to string first then check against the error name. 
```elixir
Enum.map([FunctionClauseError], & &1|> to_string() |> String.trim_leading("Elixir."))
```

File: lib/honeybadger.ex
Comment (TraceyOnim) [reply lib/honeybadger.ex:296]: Fetching the exclude_error using compile_env/2 results to the following error:
```elixir
ERROR! the application :honeybadger has a different value set for key :exclude_errors during runtime compared to compile time. Since this application environment entry was marked as compile time, this difference can lead to different behaviour than expected:

  * Compile time value was not set
  * Runtime value was set to: []

To fix this error, you might:

  * Make the runtime value match the compile time one

  * Recompile your project. If the misconfigured application is a dependency, you may need to run "mix deps.compile honeybadger --force"

  * Alternatively, you can disable this check. If you are using releases, you can set :validate_compile_env to false in your release configuration. If you are using Mix to start your system, you can pass the --no-validate-compile-env flag


** (ErlangError) Erlang error: "aborting bo
```

From the error it suggests we have set a runtime option. Does that mean we also need to set a compile option ? cc @sorentwo

File: CHANGELOG.md
Comment (joshuap) [inline CHANGELOG.md:9]: We can add the heading for this when we release (after this PR is merged).

```suggestion
```

**Review by joshuap — COMMENTED** [review]
LGTM. I realized I can't approve this PR since I'm the author. I want to wait for @sorentwo to review and clarify the compile-time question anyway. 👍

File: lib/honeybadger.ex
Comment (sorentwo) [reply lib/honeybadger.ex:296]: It would have to be referenced in the module body, outside of a function. Don't worry about it, there are several places where we're referencing the application environment during runtime. It's just not a recommended practice.

**Review by sorentwo — APPROVED** [review]

**Comment by joshuap** [comment]
@TraceyOnim This just needs documentation in the readme.
cc @subzero10

**Comment by subzero10** [comment]
> @TraceyOnim This just needs documentation in the readme. cc @subzero10

Hey @TraceyOnim, are you working on this?

**Comment by TraceyOnim** [comment]
Yes, I'm working on it

File: README.md
Comment (subzero10) [inline README.md:194]:
```suggestion
By default Honeybadger will be notified when an error occurs. To override this configuration in order not to send out errors to Honeybadger, set `exclude_errors` option in `config/config.exs`.
```

**Review by subzero10 — CHANGES_REQUESTED** [review]

**Review by subzero10 — APPROVED** [review]

**Pull request merged**
